src/expectEvent.js

```diff
@@ -26,21 +55,55 @@ function inLogs (logs, eventName, eventArgs = {}) {
 }
 
 async function inConstruction (contract, eventName, eventArgs = {}) {
+  if (!isTruffleContract(contract)) {
+    throw new Error('expectEvent.inConstruction is only supported for truffle-contract objects');
+  }
+
   return inTransaction(contract.transactionHash, contract.constructor, eventName, eventArgs);
 }
 
 async function inTransaction (txHash, emitter, eventName, eventArgs = {}) {
   const receipt = await web3.eth.getTransactionReceipt(txHash);
-  const logs = emitter.decodeLogs(receipt.logs);
+
+  const logs = decodeLogs(receipt.logs, emitter, eventName);
   return inLogs(logs, eventName, eventArgs);
 }
 
+// This decodes longs for a single event type, and returns a decoded object in
+// the same form truffle-contract uses on its receipts
+function decodeLogs (logs, emitter, eventName) {
+  let abi;
+  if (isWeb3Contract(emitter)) {
+    abi = emitter.options.jsonInterface;
+  } else if (isTruffleContract(emitter)) {
+    abi = emitter.abi;
+  } else {
+    throw new Error('Unknown contract object');
+  }
+
+  let eventABI = abi.filter(x => x.type === 'event' && x.name === eventName);
```

Comment:
I wanted to fail loudly if there multiple events with the same name in the ABI, instead of defaulting to the first one. The error message could be improved though.

Reply:
Got it. I hadn't considered that possibility. Agree on improving the error message.